File: src/main/java/net/alephdev/calendar/models/Sprint.java
package net.alephdev.calendar.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDate;
import lombok.Data;

@Data
@Entity
@Table(name = "Sprint")
public class Sprint {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private String majorVersion;

  @Column(nullable = false)
  private LocalDate startDate;

  @Column(nullable = false)
  private LocalDate endDate;

  @Column(nullable = false)
  private LocalDate regressionStart;

  @Column(nullable = false)
  private LocalDate regressionEnd;

  @ManyToOne
  @JoinColumn(
      name = "team_id",
      foreignKey = @ForeignKey(name = "fk_sprint_team_id"),
      nullable = false)
  private Team team;
}

File: angular/src/app/models/sprint.ts
import {Team} from "./team";

/**
 * Модель спринта в Agile методологии.
 *
 * Представляет временной промежуток (итерацию) в процессе разработки,
 * в течение которого команда работает над набором задач.
 *
 * Спринт имеет четко определенные даты начала и окончания,
 * а также период регрессионного тестирования.
 *
 * @property id - Уникальный идентификатор спринта
 * @property majorVersion - Версия спринта (например, "1.0.0", "2.1.5")
 * @property startDate - Дата начала спринта (ISO формат)
 * @property endDate - Дата окончания спринта (ISO формат)
 * @property regressionStart - Дата начала регрессионного тестирования (ISO формат)
 * @property regressionEnd - Дата окончания регрессионного тестирования (ISO формат)
 * @property team - Команда, работающая над спринтом
 */
export interface Sprint {
  id: number;
  majorVersion: string;
  startDate: string; // Date in ISO format
  endDate: string; // Date in ISO format
  regressionStart: string; // Date in ISO format
  regressionEnd: string; // Date in ISO format
  team: Team;
}
